allensdk/test/brain_observatory/behavior/test_stimulus_processing.py

```diff
@@ -64,13 +153,125 @@ def test_get_draw_epochs(start_frame, stop_frame, expected):
+                               "stimulus_presentations_id": [0, 1, 2, 3]})
+                         ], indirect=['behavior_stimuli_time_fixture',
+                                      'behavior_stimuli_data_fixture'])
+def test_get_stimulus_presentations(behavior_stimuli_time_fixture,
+                                    behavior_stimuli_data_fixture,
+                                    expected):
+    (behavior_stimuli_times,
+     behavior_fixture_params) = behavior_stimuli_time_fixture
+    presentations_df = get_stimulus_presentations(
+        behavior_stimuli_data_fixture,
+        behavior_stimuli_times)
+
+    presentations_df = presentations_df.drop('index', axis=1)
+
+    expected_df = pd.DataFrame.from_dict(expected)
+    expected_df.set_index('stimulus_presentations_id', inplace=True)
+    expected_df = expected_df[sorted(expected_df.columns)]
+
+    assert presentations_df.equals(expected_df)
+
+
+@pytest.mark.parametrize("behavior_stimuli_time_fixture,"
+                         "behavior_stimuli_data_fixture,"
+                         "expected_data", [
+                             ({"timestamp_count": 15, "time_step": 1},
+                              {"images_set_log": [
+                                  ('Image', 'im065', 5, 0),
+                                  ('Image', 'im064', 25, 6)
+                              ],
+                                  "images_draw_log": (([0] * 2 + [1] * 2 +
+                                                       [0] * 3) * 2 + [0]),
+                                  "grating_set_log": [
+                                      ("Ori", 90, 3.5, 0),
+                                      ("Ori", 270, 15, 6)
+                                  ],
+                                  "grating_draw_log": (([0] + [1] * 3 + [0] * 3)
+                                                       * 2 + [0])},
+                              {"orientation": [90, None, 270, None],
+                               "image_name": [None, 'im065', None, 'im064'],
+                               "frame": [2.0, 3.0, 9.0, 10.0],
+                               "end_frame": [5.0, 5.0, 12.0, 12.0],
+                               "time": [2.0, 3.0, 9.0, 10.0],
+                               "duration": [3.0, 2.0, 3.0, 2.0],
+                               "omitted": [False, False, False, False]})
+                         ],
+                         indirect=["behavior_stimuli_time_fixture",
+                                   "behavior_stimuli_data_fixture"])
+def test_get_visual_stimuli_df(behavior_stimuli_time_fixture,
+                               behavior_stimuli_data_fixture,
+                               expected_data):
+    (behavior_stimuli_times,
+     behavior_fixture_params) = behavior_stimuli_time_fixture
+    stimuli_df = get_visual_stimuli_df(behavior_stimuli_data_fixture,
+                                       behavior_stimuli_times)
+    stimuli_df = stimuli_df.drop('index', axis=1)
```

Comment:
In general, I think you should try to avoid modifying the return values you're testing against. Here, it's pretty minor, but, why not just test the index also? The future reader is left to wonder "why?" as opposed to you putting in the test what the index should be.

Reply:
Good point, I removed the modifications to the data before assertions.